types/src/error.rs

```diff
@@ -24,7 +26,7 @@ pub enum CallError {
 	InvalidParams,
 	#[error("RPC Call failed: {0}")]
 	/// The call failed.
-	Failed(#[source] Box<dyn std::error::Error + Send + Sync>),
+	Failed(JsonRpcErrorObjectOwned),
```

Comment:
Hmm. Isn't this going to make things messy in substrate? As it is now, with a `Box<std::error::Error>`, it is fairly ergonomic to work with app specific errors (as long as they implement `dyn Error` it's all good). Won't we have a lot of pesky error conversion code with this?

It is true that in some places, `substrate` does this verbose verbose error handling, e.g. [here](https://github.com/paritytech/substrate/blob/e6cd6b1240ce741f06b05ad1c59f5399e52e717a/frame/transaction-payment/rpc/src/lib.rs#L103-L112).

So yeah, maybe we have to do this? :(

Reply:
indeed, that's why we need this but if you could convince that we don't need those application specific error codes and so on, I would happy to kill this PR :)

However, each rpc module in substrate has their own specific error codes, messages and data that we don't support currently

So you are right, it will be very boiler platey users of this API I really liked what had :(

I will investigate whether a trait design is better in this case, but we provide a macro for this or something.